### PR TITLE
Header Updates

### PR DESCRIPTION
**Pull request by wilwade**
# Problem

Header needed to match the new styles

# Solution

- Does not handle the new tablet view except by switching to mobile view
- May not 100% match yet, but additional edits should become new issues (likely from trying to use it)
- Ability to highlight with section as they are in the viewport (may need more tweaking once it is in use outside of the style guide)

## Screenshots (optional):

image

image
 
image

Ability to highlight with sections

image

### CODE REVIEW

File: src/lib/assets/Assets.stories.svelte [hunk 1]
```diff
@@ -60,7 +60,8 @@
       <div class="flex flex-col items-start justify-start p-4">
         <h2 class="mb-2 text-lg font-semibold">{name}</h2>
         {#if Icon === OpenClose}
-          <aside>* Click me!</aside>
+          <input type="checkbox" class="peer hidden" id="mobile-menu-toggle" />
```
Comment (wilwade) [inline src/lib/assets/Assets.stories.svelte:63]: The open close now all triggers off of a checkbox. Why?

1. CSS rendering is always faster than js
2. Works without JS

File: src/lib/assets/logos/FrequencyLogo.svelte [hunk 2]
```diff
@@ -13,7 +8,7 @@
...
-    fill={logoHex}
+    fill="currentColor"
```
Comment (wilwade) [inline src/lib/assets/logos/FrequencyLogo.svelte:11]: Text color will now set the color of the logo

File: src/lib/atoms/Button.svelte [hunk 3]
```diff
@@ -22,7 +22,7 @@
   $: typeClass =
     type === 'primary'
       ? 'bg-teal text-navy hover:bg-tealDark hover:text-black hover:shadow-md'
-      : 'bg-transparent border border-white hover:border-navy hover:text-primary';
+      : 'bg-transparent border border-white hover:border-navy hover:text-navy';
```
Comment (wilwade) [inline src/lib/atoms/Button.svelte:25]: Not sure what primary was, but it is not a thing

File: src/lib/features/Header.svelte [hunk 4]
```diff
@@ -5,32 +5,31 @@
...
-    window.addEventListener('resize', () => {
-      innerWidth = window.innerWidth;
-      hasMobileNav = innerWidth <= 744;
-      if (!hasMobileNav) isNavOpen = false;
-    });
-  }
 </script>
 
-<div class={`flex justify-between bg-navy p-f16 md:bg-transparent md:p-0 ${$$restProps.class}`}>
+<div class={`flex h-[80px] justify-between p-f16 md:p-0 ${$$restProps.class}`}>
```
Comment (wilwade) [inline src/lib/features/Header.svelte:10]: Not a fan of the fixed header size, but it is what it is for now. Hoping to be able to remove it and let it stretch to fill

File: src/lib/features/NavMenu.svelte [hunk 5]
```diff
@@ -1,22 +1,88 @@
...
+  export let highlightMarginTop = '0px';
+
+  let viewportTrackingItems: [string, boolean][] = menuItems
+    .filter((item) => item.viewportHighlightId)
+    .map((item) => [item.viewportHighlightId as string, false]);
+
+  // Order matters, so we just want the
+  $: highlightId = viewportTrackingItems.find(([id, inViewport]) => inViewport === true)?.[0];
+
+  onMount(() => {
```
Comment (wilwade) [inline src/lib/features/NavMenu.svelte:16]: This is the magic that allows the highlight to happen if the `viewportHighlightId` is in the viewport

File: src/lib/features/NavMenu.svelte [hunk 6]
```diff
@@ -1,22 +1,88 @@
...
+    // Create intersection observer
+    const observer = new IntersectionObserver(
+      (entries) => {
+        entries.forEach((e) => {
+          const index = viewportTrackingItems.findIndex(([id]) => id === e.target.id);
+          if (index > -1) tick().then(() => (viewportTrackingItems[index][1] = e.isIntersecting));
```
Comment (wilwade) [inline src/lib/features/NavMenu.svelte:22]: `tick` is needed so that it works on initial load

**Review by claireolmstead — APPROVED** [review]
Lots of great changes in here. Committed a couple more to more tightly align it with the designs. feel free to edit however you see fit.

**Pull request merged**